File: api/submissions.js
import { createClient } from '@supabase/supabase-js'

const supabaseUrl = 'https://ehcazveenuygfvkehzwj.supabase.co'
const supabaseKey = process.env.SUPABASE_ANON_KEY || 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJzdXBhYmFzZSIsInJlZiI6ImVoY2F6dmVlbnV5Z2Z2a2VoendqIiwicm9sZSI6ImFub24iLCJpYXQiOjE3NTQyMzAwMDcsImV4cCI6MjA2OTgwNjAwN30.gl6UsgN6zYvvGUMCpg0NbM3tpG5rPmb3-C-4BcbWQXI'
const supabase = createClient(supabaseUrl, supabaseKey)

export default async function handler(req, res) {
  // Enable CORS
  res.setHeader('Access-Control-Allow-Origin', '*');
  res.setHeader('Access-Control-Allow-Methods', 'GET, OPTIONS');
  res.setHeader('Access-Control-Allow-Headers', 'Content-Type');

  if (req.method === 'OPTIONS') {
    return res.status(200).end();
  }

  if (req.method !== 'GET') {
    return res.status(405).json({ error: 'Method not allowed' });
  }

  try {
    const { form_id } = req.query;

    if (!form_id) {
      return res.status(400).json({ error: 'form_id parameter is required' });
    }

    const formId = parseInt(form_id);
    if (isNaN(formId)) {
      return res.status(400).json({ error: 'Invalid form_id parameter' });
    }

    // Get submissions for the form from Supabase
    const { data: formSubmissions, error: getError } = await supabase
      .from('submissions')
      .select('*')
      .eq('form_id', formId)
      .order('created_at', { ascending: false });
      
    if (getError) {
      console.error('Supabase submissions error:', getError);
      return res.status(500).json({ error: 'Failed to fetch submissions' });
    }
    
    return res.status(200).json(formSubmissions);

  } catch (error) {
    console.error('Submissions API Error:', error);
    return res.status(500).json({ error: `Error fetching submissions: ${error.message}` });
  }
}

File: api/submissions.py
from http.server import BaseHTTPRequestHandler
import json
import urllib.parse
from _utils import get_submissions_by_form_id

class handler(BaseHTTPRequestHandler):
    def do_GET(self):
        """Handle GET /api/submissions?form_id={id}"""
        try:
            parsed_path = urllib.parse.urlparse(self.path)
            query_params = urllib.parse.parse_qs(parsed_path.query)
            
            # Get form_id from query parameters
            if 'form_id' not in query_params:
                self._send_error(400, "form_id parameter is required")
                return
            
            try:
                form_id = int(query_params['form_id'][0])
            except (ValueError, IndexError):
                self._send_error(400, "Invalid form_id parameter")
                return
            
            # Get submissions for the form
            submissions = get_submissions_by_form_id(form_id)
            self._send_json_response(submissions)
            
        except Exception as e:
            self._send_error(500, f"Error fetching submissions: {str(e)}")
    
    def _send_json_response(self, data, status=200):
        """Send JSON response"""
        self.send_response(status)
        self.send_header('Content-Type', 'application/json')
        self.send_header('Access-Control-Allow-Origin', '*')
        self.send_header('Access-Control-Allow-Methods', 'GET, OPTIONS')
        self.send_header('Access-Control-Allow-Headers', 'Content-Type')
        self.end_headers()
        self.wfile.write(json.dumps(data).encode('utf-8'))
    
    def _send_error(self, status, message):
        """Send error response"""
        self.send_response(status)
        self.send_header('Content-Type', 'application/json')
        self.send_header('Access-Control-Allow-Origin', '*')
        self.send_header('Access-Control-Allow-Methods', 'GET, OPTIONS')
        self.send_header('Access-Control-Allow-Headers', 'Content-Type')
        self.end_headers()
        error_data = {"error": message}
        self.wfile.write(json.dumps(error_data).encode('utf-8'))
    
    def do_OPTIONS(self):
        """Handle CORS preflight requests"""
        self.send_response(200)
        self.send_header('Access-Control-Allow-Origin', '*')
        self.send_header('Access-Control-Allow-Methods', 'GET, OPTIONS')
        self.send_header('Access-Control-Allow-Headers', 'Content-Type')
        self.end_headers()